agent/testflinger_agent/runner.py

```diff
@@ -78,16 +80,16 @@ def post_output(self, data: str):
     def register_stop_condition_checker(self, checker: StopConditionType):
         self.stop_condition_checkers.append(checker)
 
-    def check_stop_conditions(self) -> str:
+    def check_stop_conditions(self) -> Tuple[Optional[TestEvent], str]:
         """
         Check stop conditions and return the reason if any are met. Otherwise,
         return an empty string if none are met
         """
         for checker in self.stop_condition_checkers:
             output = checker()
-            if output:
+            if output[0] is not None:
                 return output
```

Comment:
You still need to return 2 values here, not just 1. When we are calling self.check_stop_condition() on line 142, it's expecting both the event and the stop_reason (which should maybe be "detail" or something since now the event also conveys the reason for it)

Reply:
I think output here is two values based on the return of the stop condition checkers, but I can make that more explicit.